zillow_address_pull.py

```diff
@@ -21,17 +23,99 @@ def random_address():
+
+                # connect to Zillow API
+                zillow_api = ZillowWrapper('X1-ZWz17hux7ivthn_9myga')
+                deep_search_response = zillow_api.get_deep_search_results(address, zipcode)
+                result = GetDeepSearchResults(deep_search_response)
+
+                # pull all dimensions and measures from zillow api
+                zestimate = result.zestimate_amount
+                rent_zestimate = result.rentzestimate_amount
+                zest_last_upd = result.zestimate_last_updated
+                zest_value_chg = result.zestimate_value_change
+                zestimate_percentile = result.zestimate_percentile
+                home_type = result.home_type
+                longitude = result.longitude
+                latitude = result.latitude
+                year_built = result.year_built
+                property_size = result.property_size
+                home_size = result.home_size
+                num_of_bathrooms = result.bathrooms
+                num_of_bedrooms = result.bedrooms
+                tax_year = result.tax_year
+                tax_value = result.tax_value
+                last_sold_date = result.last_sold_date
+                last_sold_price = result.last_sold_price
+
+                # assign values
+                values = [address, zipcode, zestimate, rent_zestimate, zest_last_upd, zest_value_chg,
+                          zestimate_percentile, home_type, longitude, latitude, year_built, property_size,
+                          home_size, num_of_bathrooms, num_of_bedrooms, tax_year, tax_value, last_sold_date,
+                          last_sold_price, full_address]
+
+                # transform dataset into a series to be able to append it to the dataset.
+                a_series = pd.Series(values, index=housing_df.columns)
+                housing_df = housing_df.append(a_series, ignore_index=True)
+                print('GREAT SUCCESS')
+
+
+
+            except:
+                pass
+
+        # concat the matched_addresses
+        result = pd.concat([housing_df, matched_addresses], ignore_index=True, sort=False)
+        result.rename(columns={"full_address_x": "full_address"})
+        # result_new = result.drop(columns=['full_address_x', 'full_address_y', 'exists'], axis=1)
+        result_new = result.drop(columns=['exists'], axis=1)
+        result_new.to_excel('/home/bobby/zillow_project/housing_df.xlsx', index=False)
+
+        zestimate = result['zestimate']
+        return zestimate
+
+    else:
+        print('something')
+
+
+# Pull housing data from the Zillow API for the single address input function on flask app
+def zillow_info(address, zipcode):
     # import address list housing data, and neighborhood mapping
-    housing_data = pd.read_csv('housing_data.csv')
-    neighborhood_mapping = pd.read_csv('neighborhood mapping.csv')
+    housing_data = pd.read_csv('housing_data.xlsx')
```

Comment:
So apparently Github PR reviews are terrible and I can't comment on lines that have been collapsed. But I recommend finding some way to DRY up lines 139++ here since it seems like very similar logic to what you do in the other for loop.